src/components/WebexActivityStream/TimeRuler.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {ListSeparator} from '@momentum-ui/react';
+import {format, isSameWeek, isToday, isYesterday} from 'date-fns';
+
+import './TimeRuler.scss';
+
+/**
+ * Returns a formatted timestamp based on the given date's offset from the current time.
+ *
+ * Divisor for messages from today display today
+ * Divisor for messages from dates from a previous day display as Yesterday
+ * Divisor for messages of dates from a previous day of the week (but not yesterday) display as <day of the week>
+ * Divisor for messages of dates older than a week from today display as M/D/YY
+ *
+ * @param {Date} timestamp Date instance to format
+ * @returns {string} formattedDate
+ */
+export function formatTimeRulerText(timestamp) {
+  let formattedDate;
+
+  if (isToday(timestamp)) {
+    formattedDate = 'Today';
+  } else if (isYesterday(timestamp)) {
+    // Yesterday
+    formattedDate = 'Yesterday';
+  } else if (isSameWeek(timestamp, new Date())) {
+    // Monday
+    formattedDate = format(timestamp, 'iiii');
+  } else {
+    // 1/1/2020
+    formattedDate = format(timestamp, 'P');
+  }
+
+  return formattedDate;
+}
+
+/**
+ * Time Ruler component is used by the activity stream
+ * to separate activities from different days.
+ *
+ * @param {object} props
```

Comment:
Could you add a param for the `props.date` item?

Reply:
Is this only for this component or all of them? I left it as generic "props" for all components, hoping the `propTypes` are enough